libs/server/repository/core/domain/src/lib/utils/iterator-to-array.util.ts

```diff
@@ -0,0 +1,19 @@
+export function iteratorToArray<T>(a: Iterator<T>): T[] {
+  const result: T[] = [];
+  let current = a.next();
+  while (!current.done) {
+    result.push(current.value);
+    current = a.next();
+  }
+  return result;
+}
+
+export async function asyncIteratorToArray<T>(a: AsyncIterator<T>): Promise<T[]> {
```

Comment:
a lot of those things should be moved to util library. Under domain/utils live just things related to the domain's problem.